app/models/task.py

```diff
@@ -1,5 +1,33 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from sqlalchemy import String, Boolean, ForeignKey
+from typing import Optional
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[str | None] = mapped_column(String, nullable=True)
+    is_complete: Mapped[bool] = mapped_column(Boolean, default=False)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def to_dict(self, include_goal_id=False):
+        response = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None,
+        }
+        if include_goal_id and self.goal_id is not None:
```

Comment:
So the second condition is really the only condition we need to a check for here since if a task doesn't have a goal then `goal_id` will be `None` regardless. Also another thing to consider is the consistency of our responses. Sometimes adding a `key` (`goal_id`) and sometimes not makes our API less uniform.